presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -268,10 +272,12 @@ private Page processPageSource()
             pageSourceMemoryContext.setBytes(pageSource.getSystemMemoryUsage());
 
             if (page != null) {
+                page = recordProcessedInput(page);
+
                 // update operator stats
                 long endCompletedBytes = pageSource.getCompletedBytes();
                 long endReadTimeNanos = pageSource.getReadTimeNanos();
-                operatorContext.recordProcessedInput(endCompletedBytes - completedBytes, page.getPositionCount(), endReadTimeNanos - readTimeNanos);
+                operatorContext.recordRawInputWithTiming(endCompletedBytes - completedBytes, endReadTimeNanos - readTimeNanos);
```

Comment:
How does it work in case of a LazyBlock? Would it record only the stats for reading block footer? Shouldn't it be done when block is actually loaded?

Reply:
`pageSource.getCompletedBytes();` is monotonically increasing. Therefore we might be at most one lazy page (last) off when we account immediately after fetching lazy page. Probably `completed bytes` should also be updated one last time before finishing, but it shouldn't make huge difference.

Reply:
Thanks for the explanation